cli/lib/kontena/cli/stacks/yaml/validations.rb

```diff
@@ -43,7 +43,7 @@ def common_validations
               )
             end
           elsif value.is_a?(Array)
-            value.all? { |v| v.kind_of?(String) && v =~ /\A\S+(?<!\\)=.*/ }
+            value.all? { |v| v.kind_of?(String) && v =~ /^([^=]+)=(.*)/ }
```

Comment:
Right, it gets a little crazy if there's newlines in the env string..

```ruby
irb(main):005:0> /^([^=]+)=(.*)/.match "ASDF\nFOO=bar\nquux"
=> #<MatchData "ASDF\nFOO=bar" 1:"ASDF\nFOO" 2:"bar">
irb(main):006:0> /\A([^=]+)=(.*)/.match "ASDF\nFOO=bar\nquux"
=> #<MatchData "ASDF\nFOO=bar" 1:"ASDF\nFOO" 2:"bar">
```

What about `/\A\S+=(.*)\Z/`?

```
irb(main):017:0> /\A\S+=(.*)\Z/.match "ASDF\nFOO=bar\nQUUX"
=> nil
irb(main):018:0> /\A\S+=(.*)\Z/.match "FOO=bar\nQUUX"
=> nil
irb(main):019:0> /\A\S+=(.*)\Z/.match "ASDF\nFOO=bar"
=> nil
irb(main):020:0> /\A\S+=(.*)\Z/.match "FOO=bar"
=> #<MatchData "FOO=bar" 1:"bar">
```